Nakama/WebSocketAdapter.cs

```diff
@@ -86,6 +86,7 @@ public Task CloseAsync()
             if (_webSocket == null) return Task.CompletedTask;
             var t = _webSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, "", CancellationToken.None);
             _webSocket = null;
+            Closed?.Invoke();
```

Comment:
@csdaraujo I think this already should fire in our `ReceiveLoop` -- is there a particular test that was failing that caused you to make this change?

Reply:
`ShouldCreateSocketAndDisconnect` was failing.
And `ConnectAsync` function has the same approach to the action (in that case `Connected?.Invoke();`)
But I can change if this other approach is the way to go.